documentation/projects/proposals/ingestion_server_removal/20240319-project_proposal.md

```diff
@@ -0,0 +1,105 @@
+# 2024-03-19 Project Proposal: Removal of the Ingestion Server
+
+**Author**: @stacimc
+
+## Reviewers
+
+<!-- Choose two people at your discretion who make sense to review this based on their existing expertise. Check in to make sure folks aren't currently reviewing more than one other proposal or RFC. -->
+
+- [ ] @Aetherunbound
+- [ ] @zackkrida
+
+## Project summary
+
+<!-- A brief one or two sentence summary of the project's features -->
+
+Currently, our data refresh process is _orchestrated_ by Airflow, but the work
+is actually performed on a bespoke ingestion server. This server is relatively
+complex compared to our needs, difficult to maintain, and makes the data refresh
+process unnecessarily opaque. Because the refresh is triggered in a single task
+on the ingestion server, it is not possible to track the progress in Airflow or
+retry it from a point of failure. Progress must be followed through periodic
+Slack notifications and inspection of logs in Cloudwatch.
+
+In this project, we will move all operational pieces of the data refresh into
+Airflow and completely remove the ingestion server.
+
+## Goals
+
+<!-- Which yearly goal does this project advance? -->
+
+Moving the process entirely into Airflow will have the following advantages for
+the critical data refresh process:
+
+- Easier iteration, as changes to the implementation can be deployed immediately
+  by simply merging code in the catalog
+- Increased reusability of code, particularly among the growing number of
+  Elasticsearch DAGs
```

Comment:
Would you mind providing an example here? It's the result of my own ignorance, but I'm not able to think of one independently and would find it useful.

Reply:
Will do -- these were added in the last several months as part of the Search Relevancy Sandbox project :) One example is the `recreate_full_staging_index` DAG which will likely be made totally redundant once the data refresh is in Airflow.